libvast/src/system/exporter.cpp

```diff
@@ -108,6 +107,13 @@ behavior exporter(stateful_actor<exporter_state>* self, expression expr,
       self->quit(msg.reason);
     }
   );
+  self->set_down_handler(
+    [=](const down_msg& msg) {
+      VAST_DEBUG(self, "received DOWN from", msg.source);
+      if (msg.reason == exit_reason::user_shutdown)
+        shutdown(self);
```

Comment:
A follow up question (to be sure): CTRL + C'ing the SINK process (i.e. `vast export -c "..."`) prompts the NODE shutdown related the EXPORTERs with a `down_msg`?

When should shutdown be called here. Previously, this wasn't called when a `down_msg` or `exit_msg` was received. For the continuous query, shutdown should be called when the SINK is shutdown by the user. To prevent confusion with the other cases where shutdown is called, we could also extract the reporting functionality from `shutdown`  and call that here ...

```cpp
void report(stateful_actor<exporter_state>* self) {
  timespan runtime = steady_clock::now() - self->state.start;
  self->state.stats.runtime = runtime;
  VAST_DEBUG(self, "completed in", runtime);
  self->send(self->state.sink, self->state.id, self->state.stats);
  if (self->state.accountant) {
    auto hits = rank(self->state.hits);
    auto processed = self->state.stats.processed;
    auto shipped = self->state.stats.shipped;
    auto results = shipped + self->state.results.size();
    auto selectivity = double(results) / hits;
    self->send(self->state.accountant, "exporter.hits", hits);
    self->send(self->state.accountant, "exporter.processed", processed);
    self->send(self->state.accountant, "exporter.results", results);
    self->send(self->state.accountant, "exporter.shipped", shipped);
    self->send(self->state.accountant, "exporter.selectivity", selectivity);
    self->send(self->state.accountant, "exporter.runtime", runtime);
  }
}

void shutdown(stateful_actor<exporter_state>* self) {
  if (rank(self->state.unprocessed) > 0 || !self->state.results.empty()
      || has_continuous_option(self->state.opts))
    return;
  report(self);
  self->send_exit(self, exit_reason::normal);
}
```